scripts/build-all-dev-images

```diff
@@ -21,13 +21,13 @@ for python in 3.8 3.9 3.10; do
 
         echo
         echo "---- expecting Python $python$flavor_desc ----"
-        docker run --rm prefecthq/prefect:dev-python$python$flavor_tag python --version
+        docker run --rm prefecthq/prefect-dev:sha-$(git rev-parse --short HEAD)-python$python$flavor_tag python --version
```

Comment:
You use `git rev-parse --short=7 HEAD` in `.github/workflows/python-tests.yaml`. Do we want the same sha length here?
```suggestion
        docker run --rm prefecthq/prefect-dev:sha-$(git rev-parse --short=7 HEAD)-python$python$flavor_tag python --version
```

Reply:
Thanks!